addons/pos_sale/models/pos_order.py

```diff
@@ -46,6 +46,15 @@ def _prepare_invoice_vals(self):
                 invoice_vals['partner_id'] = sale_orders[0].partner_invoice_id.id
         return invoice_vals
 
+    def action_pos_order_paid(self):
+        res = super().action_pos_order_paid()
+        if any(p.payment_method_id._is_online_payment() for p in self.payment_ids):
+            sale_orders = (self.lines.mapped('sale_order_line_id.order_id') or
```

Comment:
I don't think we need to have a fallback here we can just use sale_order_origin_id. Also you should probably use a set to avoid doing `action_confirm` multiple times on the same sale order